connectors/src/connectors/confluence/index.ts

```diff
@@ -465,8 +469,33 @@ export class ConfluenceConnectorManager extends BaseConnectorManager<null> {
     return new Ok(contentNodes);
   }
 
+  private _cachedHierarchyMutex = new Mutex();
+  private _cachedHierarchy: Map<string, Map<string, string | null>> = new Map();
+
+  private async getCachedHierarchy(
+    memoizationKey: string,
+    spaceId: string
+  ): Promise<Map<string, string | null>> {
+    return this._cachedHierarchyMutex.runExclusive(async () => {
+      if (!this._cachedHierarchy.has(memoizationKey)) {
+        this._cachedHierarchy.set(
+          memoizationKey,
+          await getSpaceHierarchy(this.connectorId, spaceId)
+        );
+      }
+
+      // pleasing typescript
+      const cachedHierarchy = this._cachedHierarchy.get(memoizationKey);
+      if (!cachedHierarchy) {
+        throw new Error("Unreachable: Cached hierarchy is not set.");
+      }
```

Comment:
`nit`
You can use `assert` for this.

Reply:
I used to do that and was rebuked in PR :sweat_smile: Glad to migrate back to it going forward :)